nav2_toolkit/test/test_pose_saver_node.cpp

```diff
@@ -0,0 +1,257 @@
+// Copyright (c) 2025 Pranav Kolekar
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <chrono>
+#include <filesystem>
+#include <random>
+#include <thread>
+#include <rclcpp/rclcpp.hpp>
+#include <geometry_msgs/msg/pose_with_covariance_stamped.hpp>
+#include <std_srvs/srv/trigger.hpp>
+#include "nav2_toolkit/pose_saver_node.hpp"
+
+using nav2_toolkit::PoseSaverNode;
+namespace fs = std::filesystem;
+
+// Test fixture that inherits from PoseSaverNode to access protected members
+class TestablePoseSaverNode : public PoseSaverNode
```

Comment:
https://app.codecov.io/gh/ros-navigation/navigation2/pull/5369?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=checks&utm_campaign=pr+comments&utm_term=ros-navigation 

we're missing alot of test coverage, new packages must be > 90% and should be generally at 95% or higher